src/ensembl/src/content/app/custom-download/components/content-builder/ContentBuilder.tsx

```diff
@@ -0,0 +1,263 @@
+import {
+  Accordion,
+  AccordionItem,
+  AccordionItemHeading,
+  AccordionItemPanel,
+  AccordionItemButton
+} from 'src/shared/accordion';
+
+import styles from './ContentBuilder.scss';
+
+import get from 'lodash/get';
+import {
+  Attribute,
+  AttributeWithContent,
+  AttributeWithOptions
+} from 'src/content/app/custom-download/types/Attributes';
+import JSONValue, { PrimitiveOrArrayValue } from 'src/shared/types/JSON';
+
+type Path = (string | number)[];
+let path: Path = [];
+
+type ContentBuilderProps = {
+  data: AttributeWithContent;
+  selectedData: JSONValue;
+  onChange: (type: string, path: Path, payload: PrimitiveOrArrayValue) => void;
+  contentState: JSONValue;
+  onContentStateChange: (
+    type: string,
+    path: Path,
+    payload: PrimitiveOrArrayValue
+  ) => void;
+  path?: Path;
+  contentProps?: {
+    [key: string]: JSONValue;
+  };
+};
+
+const ContentBuilder = (props: ContentBuilderProps) => {
+  const onChangeHandler = (
+    type: string,
+    path: Path,
+    payload: PrimitiveOrArrayValue
+  ) => {
+    props.onChange(type, path, payload);
+  };
+
+  const onContentStateChangeHandler = (
+    type: string,
+    path: Path,
+    payload: PrimitiveOrArrayValue
+  ) => {
+    props.onContentStateChange(type, path, payload);
+  };
+
+  const buildCheckboxWithSelect = (entry: AttributeWithOptions, path: Path) => {
+    const currentPath = [...path, entry.id];
+
+    const selectedOptions = get(props.selectedData, currentPath, []);
+
+    return (
+      <div className={styles.checkboxWithSelectWrapper}>
+        <CheckboxWithSelects
+          label={entry.label}
+          disabled={entry.disabled}
+          onChange={(selectedOptions: string[]) =>
+            onChangeHandler(entry.type, currentPath, selectedOptions)
+          }
+          selectedOptions={selectedOptions}
+          options={(entry.options as Option[]) || []}
+        />
+      </div>
+    );
+  };
+
+  const buildCheckboxWithRadios = (entry: AttributeWithOptions, path: Path) => {
+    const currentPath = [...path, entry.id];
+
+    const selectedOption: string = get(props.selectedData, currentPath, '');
+
+    return (
+      <div className={styles.checkboxWitRadiosWrapper}>
+        <CheckboxWithRadios
+          label={entry.label}
+          disabled={entry.disabled}
+          onChange={(selectedOption: string | number | boolean) =>
+            onChangeHandler(entry.type, currentPath, selectedOption)
+          }
+          selectedOption={selectedOption}
+          options={entry.options as RadioOptions}
+        />
+      </div>
+    );
+  };
+
+  const buildCheckboxGrid = (entry: AttributeWithOptions, path: Path) => {
+    const currentPath = [...path, entry.id];
+
+    const selectedOptions = get(props.selectedData, currentPath, []);
+
+    const gridOptions = [...entry.options];
+
+    const gridClone: CheckboxGridOption[] = [];
+
+    Object.values(gridOptions as CheckboxGridOption[]).map((option) => {
```

Comment:
Why are you running Object.values on an array? :scream_cat:

Reply:
Forget to remove it. It was an object before and then I changed it to be an array.